src/mpid/ch4/shm/xpmem/xpmem_recv.h

```diff
@@ -68,40 +66,28 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_handle_lmt_coop_recv(uint64_t src_offse
     if (src_data_sz > data_sz)
         rreq->status.MPI_ERROR = MPI_ERR_TRUNCATE;
 
-    /* XPMEM internal info */
-    slmt_cts_hdr->dest_offset = (uint64_t) MPIDIG_REQUEST(rreq, buffer) + dt_true_lb;
-    slmt_cts_hdr->data_sz = recv_data_sz;
-    slmt_cts_hdr->dest_lrank = MPIDI_XPMEM_global.local_rank;
+    /* allocate shm counter */
+    MPIDI_XPMEM_cnt_t *counter_ptr;
+    counter_ptr = (MPIDI_XPMEM_cnt_t *) MPIR_Handle_obj_alloc(&MPIDI_XPMEM_cnt_mem);
+    MPIDI_XPMEM_REQUEST(rreq, counter_ptr) = counter_ptr;
 
```

Comment:
It is very hard to figure out what is changed or moved to a different routine. Can you please avoid unnecessary change such as 
```
MPIDI_XPMEM_REQUEST(rreq, counter_ptr) =
        (MPIDI_XPMEM_cnt_t *) MPIR_Handle_obj_alloc(&MPIDI_XPMEM_cnt_mem);
```
->
```
    MPIDI_XPMEM_cnt_t *counter_ptr;
    counter_ptr = (MPIDI_XPMEM_cnt_t *) MPIR_Handle_obj_alloc(&MPIDI_XPMEM_cnt_mem);
    MPIDI_XPMEM_REQUEST(rreq, counter_ptr) = counter_ptr;
```

Reply:
sure.